accepted/2021/System.Diagnostics/Metrics-Design.md

```diff
@@ -0,0 +1,484 @@
+# Metrics APIs Design
+
+## Overview
+
+This document is discussing the .NET Metrics APIs design which implements the [OpenTelemetry specification](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/api.md).
```

Comment:
How should I think about these APIs vs the DiagnosticCounter APIs we've been using from System.Diagnostics.Tracing?  When does someone choose these vs EventCounter, PollingCounter, etc.?

Reply:
We already working on publishing the guide for the users currently using EventCounters. That include, having the tools like dotnet-counters work with new Metrics, migrating from EventCounter to Metrics, and allow migrated libraries/apps from EventCounters to Metrics to implement a small shim for any app compat support if necessary. 

CC @noahfalk @shirhatti

Reply:
Does that mean we're deprecating DiagnosticCounter, changing all of our existing DiagnosticCounters to be based on these new APIs, etc.?  I'm confused.

Reply:
We'll recommend using Metrics moving forward but we are not deprecating the Event counters in this release at least.

Reply:
So I hope then we're replacing our own usage in .NET 6?